app/main.py

```diff
@@ -1 +1,23 @@
-# write your code here
+from __future__ import annotations
+
+from app.cafe import Cafe
+from app.errors import (VaccineError, NotWearingMaskError)
+
+
+def go_to_cafe(friends: list, cafe: Cafe) -> str:
+    masks_to_buy = 0
+    access_granted = True
+    for friend in friends:
+        try:
+            cafe.visit_cafe(friend)
+        except VaccineError:
+            access_granted = False
+            return "All friends should be vaccinated"
+        except NotWearingMaskError:
+            access_granted = False
+            masks_to_buy += 1
+
+    if masks_to_buy > 0:
+        return f"Friends should buy {masks_to_buy} masks"
+    if access_granted:
```

Comment:
this will work but tbh i'm not sure you need the access_granted variable because it just does unnecessary double-checks: if VaccineError thrown - first return works, if mask_to_buy check > 0 - second return works, if both checks are passed then why would you need one more?

Reply:
Agree, remove redundant check and variable